templates/shared/issueicon.tmpl

```diff
@@ -1,16 +1,12 @@
 {{if .IsPull}}
 	{{if and .PullRequest .PullRequest.HasMerged}}
 		{{svg "octicon-git-merge" 16 "text purple"}}
-	{{else if and (.GetPullRequest ctx) (.GetPullRequest ctx).HasMerged}}
-		{{svg "octicon-git-merge" 16 "text purple"}}
```

Comment:
I am not sure whether this change could be covered by all cases: in all cases, the "PullRequest" is loaded correctly.

Reply:
To make code stable, I think here should add a "if not .PullRequest" check, and output "No PullRequest", to find problems ASAP.

Reply:
Maybe we can just remove `.PullRquest` check so that it will panic if .PullRequest haven't been loaded.

Reply:
> Maybe we can just remove `.PullRquest` check so that it will panic if .PullRequest haven't been loaded.

My solution: users only see "No PullRequest", the UI still works and users could continue working.

Your solution: users see 500 panic and can't continue working.

Reply:
> > Maybe we can just remove `.PullRquest` check so that it will panic if .PullRequest haven't been loaded.
> 
> My solution: users only see "No PullRequest", the UI still works and users could continue working.
> 
> Your solution: users see 500 panic and can't continue working.

If it's a pull request and PullRequest is nil, the pull request lists will panic in many places. Not only the issue icon.

Reply:
I could agree, but since you would like to backport, I think it shouldn't panic for end users, even if there is something wrong.

It could panic in 1.22 or dev mode.

Reply:
> I could agree, but since you would like to backport, I think it shouldn't panic for end users, even if there is something wrong.
> 
> It could panic in 1.22 or dev mode (in a separate PR)

It will already panic when loading WIP, headrepo & headbranch which will be displayed in pulls list page. So if we want to avoid that panic, we have to check all these places. I don't think it's necessary because it's a database broken problem, returning a panic could let then know the serious situation.

Reply:
Yup, you fixed a database broken bug.

But, you might introduce some new panic bug if the "PullRequest" is not prepared properly in code, that what I said in the first comment:

> I am not sure whether this change could be covered by all cases: in all cases, the "PullRequest" is loaded correctly.


If you can't guarantee that the ".PullRequest" is properly loaded everywhere, then you introduce a new panic in this backport PR.

----

That's my opinion and final review, and I won't block it if you prefer to do that.

Reply:
5ece240